src/utils/redirects.js

```diff
@@ -42,4 +42,21 @@ const getRedirect = (host, pathname, { req, dev } = {}) => {
   return matchRedirectList(host, pathname)
 }
 
+const handleFrontRedirect = (host, pathname, clickEvent) => {
```

Comment:
do we run it for every link of the website? just to make sure that we run it for link inside docs (especially those that we generate with plugins)

Reply:
It handles it two places: in the Link component when user clicks on link (before changing route) and as fallback after changing url (404 can be showed in this case, but then user will be redirected to correct url). I replaced all links in the generated pages from markdown to the Link component, so there it will be handled as first variant.

Reply:
> 404 can be showed in this case, but then user will be redirected to correct url)

yep, I saw this - does not look nice. Could you clarify this part `fallback after changing url`?

Reply:
We can intercept url change only with `Link` component, or handle it after location change. I tried to replace all direct `a` usages to `Link` component. But if somewhere still there is any `a` - it will be handled after front location change